### PR TITLE
feat: navigate to task details when clicking on task

### LINKED ISSUE #147: Fix the opening of task details from member site
### Issue Description

 when we go to the Active Contribution tab, it opens a specific member's contributions to RDS, but when I click on the link of that issue, nothing opens, it should open the GitHub issue, or take the details page of the status site.
### Expected Behavior

When a user clicks on the task link it should take the use to task details page.

### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1.  Go to member side. Click on any card.
2.  Open Contributions list click on any link.

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [x] Medium
 - [ ] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [ ] I have searched for similar issues before creating this one.
- [x] I have provided all the necessary information to understand and reproduce the issue.
- [x] I am willing to contribute to the resolution of this issue.

### PR DESCRIPTION
**Pull request by ashifkhn**
Date: 

Developer Name: Ashif

---

## Issue Ticket Number

- Closes #147 

## Description

When a user clicks on a task item in the member site, the interface should trigger a navigation event that directs the user to the detailed view of the selected task. This view will provide comprehensive information about the task, including its description, status, due date, assigned users, and any related subtasks or comments.

### Documentation Updated?

- [ ] Yes
- [x] No

### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Breaking Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
<details>
<summary>Screenshot 1</summary>

</details>

https://github.com/user-attachments/assets/80c6574e-ba07-4e80-bfa5-3dcf64c5bb60

## Test Coverage
<details>
<summary>Screenshot 1</summary>

Screenshot 2024-08-20 at 11 48 40 AM

</details>

### CODE REVIEW

File: src/components/MemberContribution/ContributionAccordianItem.tsx [hunk 1]
```diff
@@ -1,23 +1,25 @@
...
-import { Button, Link } from '@chakra-ui/react';
 import styles from './memberContribution.module.css';
 import Image from 'next/image';
 import { useGetIsSuperUser } from '../../utils/customHooks';
+import {ContributionAccordionItemProps} from "./types"
+
+
+const URL =
+  process.env.NEXT_PUBLIC_TASK_BASE_URL ??
+  'https://status.realdevsquad.com/tasks/';
```
Comment (yesyash) [inline src/components/MemberContribution/ContributionAccordianItem.tsx:15]: - is the default value required here? instead of hardcoding the value if we can ensure the value is set when creating a build does it solve the issue?

File: src/components/MemberContribution/ContributionAccordianItem.tsx [hunk 2]
```diff
@@ -44,7 +46,17 @@ export default function ContributionAccordianItem({
...
+      <h3
+        className={styles.memberContribution_taskHeading}
+      >
+        {task.id ? (
+          <Link href={`${URL}${task.id}`} isExternal>
+            {taskTitle}
+          </Link>
+        ) : (
+          <Link href={`${task.url}`} isExternal>{taskTitle}</Link>
+        )}
```
Comment (yesyash) [inline src/components/MemberContribution/ContributionAccordianItem.tsx:58]: - should we render the link directly instead of nesting it inside an h3?

File: src/components/MemberContribution/types.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  purpose:string
+  id:string
+  isNoteworthy:string
+}
+
+export type ContributionAccordionItemProps = Readonly<{
+  task: TaskDetails;
+  isTitle: boolean;
+  openTaskStatusUpdateModal: (taskId: string, isTaskNoteworthy: string) => void;
+}>;
```
Comment (yesyash) [inline src/components/MemberContribution/types.ts:18]: - if the props are specific to `ContributionAccordianItem` component let's keep it together with the component

File: tsconfig.json [hunk 4]
```diff
@@ -16,7 +16,8 @@
     "incremental": true,
     "baseUrl": "./",
     "paths": {
-      "@/*": ["./*"]
+      "@/*": ["./*"],
+      "react": ["./node_modules/@types/react"]
```
Comment (yesyash) [inline tsconfig.json:20]: - why is this required?

**Review by yesyash — DISMISSED** [review]

File: src/components/MemberContribution/ContributionAccordianItem.tsx (hunk 1 above)
Comment (ashifkhn) [reply src/components/MemberContribution/ContributionAccordianItem.tsx:15]: Yes right, this is not needed here. We can add it in prod env.

File: src/components/MemberContribution/ContributionAccordianItem.tsx (hunk 2 above)
Comment (ashifkhn) [reply src/components/MemberContribution/ContributionAccordianItem.tsx:58]: Yes removed h3.

File: src/components/MemberContribution/types.ts (hunk 3 above)
Comment (ashifkhn) [reply src/components/MemberContribution/types.ts:18]: I thought the code would look cleaner this way.  Keeping it in the same file for noe

**Review by Achintya-Chatterjee — APPROVED** [review]

File: tsconfig.json (hunk 4 above)
Comment (ankushdharkar) [reply tsconfig.json:20]: Please respond to the question first, before marking this as resolved @ashifkhn.

**Review by vinit717 — APPROVED** [review]

**Review by satyam73 — APPROVED** [review]
